src/mqttproto/_types.py

```diff
@@ -445,35 +458,61 @@ class Subscription:
     retain_handling: RetainHandling = field(
         kw_only=True, default=RetainHandling.SEND_RETAINED
     )
+    subscription_id: int = field(kw_only=True, default=0)
     group_id: str | None = field(init=False, default=None)
     _parts: tuple[str, ...] = field(init=False, repr=False, eq=False)
+    _prefix: str | None = field(init=False, repr=False, eq=False)
+    _only_hash: bool = field(init=False, repr=False, eq=False, default=False)
 
     def __attrs_post_init__(self) -> None:
-        self._parts = tuple(self.pattern.split("/"))
-        for i, part in enumerate(self._parts):
-            if "+" in part and len(part) != 1:
-                # MQTT-4.7.1-2
-                raise InvalidPattern(
-                    "single-level wildcard ('+') must occupy an entire level of the "
-                    "topic filter"
-                )
+        parts = tuple(self.pattern.split("/"))
+        prefix: str | None = ""
+        n_chop = 0
+        for i, part in enumerate(parts):
+            if "+" in part:
+                if prefix:
+                    prefix += "/"
+                if len(part) != 1:
```

Comment:
Add a blank line between unrelated branching statements.
```suggestion

                if len(part) != 1:
```